lib/counter_culture/counter.rb

```diff
@@ -130,6 +137,47 @@ def counter_cache_name_for(obj)
       end
     end
 
+    # Gets all the columns that are needed to change defined in column_names
+    #
+    # obj: object to calculate the counter cache name
+    def counter_cache_column_names_from_scope(obj) 
+      column_names = []
+      @column_names.each do |scope, column_name|
+        column_names << column_name if scope.exists?(obj.id) || obj.instance_eval(sql_query_to_eval_query(obj, scope.to_sql))
+      end
+      column_names
+    end
+
+    # Parse sql query with AND, OR, IN to a logical expression
+    #
+    # sql: the query to parse
+    def sql_query_to_eval_query(obj, sql)
```

Comment:
I am not onboard with this approach. As you say, it's not neat. But more importantly a scope can be any set of valid SQL conditions—not just the ones you cover here. And given different databases there's really an unbounded set of allowable scope queries so I don't think it's even possible to cover them all here.

I think I understand the reason you are doing this here—to check the `previous_model` against the scope. The only other approach I can come up with here to cover that would be to actually save the `previous_model` to the database, check against it and then roll back. But I don't like that approach either. It assumes you actually have the ability to roll back, which is not always a given, plus it probably has terrible performance impacts.